test_rclcpp/test/test_waitable.cpp

```diff
@@ -55,19 +55,20 @@ class WaitableWithTimer : public rclcpp::Waitable
   }
 
   void
-  add_to_wait_set(rcl_wait_set_t * wait_set) override
+  add_to_wait_set(rcl_wait_set_t & wait_set) override
   {
-    rcl_ret_t ret = rcl_wait_set_add_timer(wait_set, timer_.get(), &timer_idx_);
+    rcl_ret_t ret = rcl_wait_set_add_timer(&const_cast<rcl_wait_set_t &>(wait_set),
```

Comment:
At first glance, this looks suspicious.  We are taking an `rcl_wait_set_t &`, and then const-casting it to a `rcl_wait_set_t &`, and then taking the address of it.  But that seems unnecessary; can this be:

```suggestion
    rcl_ret_t ret = rcl_wait_set_add_timer(&wait_set,
```